src/components/Icon/index.tsx

```diff
@@ -43,68 +43,67 @@ type IconBaseProps = {
     /** Determines how the image should be resized to fit its container */
     contentFit?: ImageContentFit;
 };
-type IconProps = IconBaseProps & WithThemeStylesProps & WithThemeProps & WithStyleUtilsProps;
-
-// We must use a class component to create an animatable component with the Animated API
-// eslint-disable-next-line react/prefer-stateless-function
-class Icon extends PureComponent<IconProps> {
-    // eslint-disable-next-line react/static-property-placement
-    public static defaultProps: Partial<IconBaseProps> = {
-        width: variables.iconSizeNormal,
-        height: variables.iconSizeNormal,
-        fill: undefined,
-        small: false,
-        inline: false,
-        additionalStyles: [],
-        hovered: false,
-        pressed: false,
-        testID: '',
-        contentFit: 'cover',
-    };
-
-    render() {
-        const width = this.props.small ? variables.iconSizeSmall : this.props.width;
-        const height = this.props.small ? variables.iconSizeSmall : this.props.height;
-        const iconStyles = [this.props.StyleUtils.getWidthAndHeightStyle(width ?? 0, height), IconWrapperStyles, this.props.themeStyles.pAbsolute, this.props.additionalStyles];
-
-        if (this.props.inline) {
-            return (
-                <View
-                    testID={this.props.testID}
-                    style={[this.props.StyleUtils.getWidthAndHeightStyle(width ?? 0, height), this.props.themeStyles.bgTransparent, this.props.themeStyles.overflowVisible]}
-                >
-                    <View style={iconStyles}>
-                        <ImageSVG
-                            src={this.props.src}
-                            width={width}
-                            height={height}
-                            fill={this.props.fill}
-                            hovered={this.props.hovered}
-                            pressed={this.props.pressed}
-                            contentFit={this.props.contentFit}
-                        />
-                    </View>
-                </View>
-            );
-        }
 
+function Icon({
+    src,
+    width = variables.iconSizeNormal,
+    height = variables.iconSizeNormal,
+    fill = undefined,
+    small = false,
+    inline = false,
+    additionalStyles = [],
+    hovered = false,
+    pressed = false,
+    testID = '',
+    contentFit = 'cover',
+}: IconProps) {
+    const theme = useTheme();
+    const StyleUtils = useStyleUtils();
+    const styles = useThemeStyles();
+    const iconWidth = small ? variables.iconSizeSmall : width;
+    const iconHeight = small ? variables.iconSizeSmall : height;
+    const iconStyles = [StyleUtils.getWidthAndHeightStyle(width ?? 0, height), IconWrapperStyles, styles.pAbsolute, additionalStyles];
+    const iconFill = fill ?? theme.icon;
```

Comment:
Any reason for fallback to `theme.icon`? This caused regression:
https://github.com/Expensify/App/pull/33077#issuecomment-1871812772

Can you please raise quick PR fixing this? I don't want this PR reverted again 🙏